### PR TITLE
Add prefix + suffix functions with tests

### PR DESCRIPTION
**Pull request by pierroberto**
Add two functions to `string` module:  `prefix` and`suffix`.

### CODE REVIEW

File: src/string.ts [hunk 1]
```diff
@@ -12,4 +12,8 @@ export const capitalize = <S extends string>(s: S): Capitalize<S> =>
 export const uncapitalize = <S extends string>(s: S): Uncapitalize<S> =>
   (S.isEmpty(s) ? s : s[0].toLowerCase() + s.slice(1)) as Uncapitalize<S>
 
+export const prefix = (prefix: string) => (s: string) => `${prefix}${s}`
+
+export const suffix = (suffix: string) => (s: string) => `${s}${suffix}`
```
Comment (davidecaruso) [inline src/string.ts:17]: There are too many ways to concatenate strings, and template-string is a good one, but what would you say if we considered string semigroups of `fp-ts` to get the same result without worrying about how?

Comment (xzhavilla) [inline src/string.ts:17]: May I suggest using `<S extends string>` to return literal types instead of strings?

Comment (pierroberto) [reply src/string.ts]: I think there are a couple of solutions.

- Template literal as return type: 
```ts
declare const suffix: <A extends string, B extends string>(
  string: A,
) => (suffix: B) => `${A}${B}`

```
- Using the same signature above with no return type specified. The return type will be `string`.

I'd go for the first solution.

Any other idea?

Comment (xzhavilla) [reply src/string.ts]: Why declaring the `B` generic in the first function, if it's used in the second one? That kills inference.

Comment (pierroberto) [reply src/string.ts]: That's right, moving the declaration of `B` in the second function fixes the problem of inference.

File: package.json [hunk 2]
```diff
@@ -100,5 +100,5 @@
       "optional": true
     }
   },
-  "packageManager": "yarn@3.2.0"
+  "packageManager": "yarn@3.2.3"
```
Comment (pierroberto) [inline package.json:103]: @xzhavilla I updated yarn to its latest stable version. CI seems to be working now.

**Pull request merged**